src/core/Akka/Actor/TerminationSignalHandler.cs

```diff
@@ -0,0 +1,117 @@
+//-----------------------------------------------------------------------
+// <copyright file="TerminationSignalHandler.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2022 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2025 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Threading;
+#if NET6_0_OR_GREATER
+using System.Runtime.InteropServices;
+#endif
+
+namespace Akka.Actor
+{
+    /// <summary>
+    /// Abstraction for handling process termination signals (SIGTERM, SIGHUP, ProcessExit).
+    /// Required for .NET 10 compatibility where ProcessExit no longer fires on SIGTERM.
+    /// </summary>
+    internal interface ITerminationSignalHandler : IDisposable
+    {
+        /// <summary>
+        /// Registers a callback to be invoked when a termination signal is received.
+        /// The callback will only be invoked once, even if multiple signals are received.
+        /// </summary>
+        /// <param name="onTerminationSignal">The callback to invoke on termination signal.</param>
+        void Register(Action onTerminationSignal);
+    }
+
+#if NET6_0_OR_GREATER
+    /// <summary>
+    /// .NET 6+ implementation using PosixSignalRegistration for proper signal handling.
+    /// Handles SIGTERM and SIGHUP signals, plus ProcessExit as fallback.
+    /// This is required for .NET 10 compatibility where ProcessExit no longer fires on SIGTERM.
+    /// </summary>
+    internal sealed class PosixTerminationSignalHandler : ITerminationSignalHandler
```

Comment:
Dumb question on my part so I feel like I must be missing it - what do we do for Windows?